### DISCUSSION TITLE
detail::fast_iterator might satisfy ont of the iterator concepts from the STL

### DISCUSSION THREAD

**Discussion by Simon-Lopez** [wjakob/nanobind#116] [Show and tell]
Dear all,

With a few additional lines the `detail::fast_iterator` could satisfy `std::forward_iterator` concept.
(cf. a proposal commit [here](https://github.com/wjakob/nanobind/compare/master...Simon-Lopez:nanobind:fast_iterator_satisfies_forward_iterator_concept)).

This would then make for example `nanobind::list` satisfying the std::ranges::range concept and we could use on `nanobind::list` all the goodies that come with std::ranges (or range-v3).

More explicitely the following code fails on my machine (gcc 11.3) without the proposed modifications:

```cpp
#include <iterator>

#include <nanobind/nanobind.h>

NB_MODULE(check, module) {

static_assert(std::forward_iterator<nanobind::detail::fast_iterator>);

}
```

**Comment by wjakob** [comment]
This commit was merged in the meantime, thanks for the PR.
